File: src/access-control/alwaysAllowDocumentAccessPolicyFactory.ts
import { RawDocument } from "../document/document.js";
import { DocumentAccessPolicyFactory } from "./documentAccessPolicyFactory.js";
import { AlwaysAllowAccessPolicy } from "./policies/alwaysAllowAccessPolicy.js";
import { ResourceAccessPolicy } from "./resourceAccessPolicy.js";

export class AlwaysAllowDocumentAccessPolicyFactory
  implements DocumentAccessPolicyFactory
{
  constructor() {}

  async getAccessPolicies(
    _rawDocument: RawDocument,
  ): Promise<ResourceAccessPolicy[]> {
    return [new AlwaysAllowAccessPolicy()];
  }
}

File: python/src/semantic_retrieval/access_control/always_allow_document_access_policy_factory.py
from typing import List

from semantic_retrieval.access_control.document_access_policy_factory import (
    DocumentAccessPolicyFactory,
)
from semantic_retrieval.access_control.policies.always_allow_access_policy import (
    AlwaysAllowAccessPolicy,
    ResourceAccessPolicy,
)
from semantic_retrieval.document.document import RawDocument


class AlwaysAllowDocumentAccessPolicyFactory(DocumentAccessPolicyFactory):
    def __init__(self):
        pass

    async def get_access_policies(
        self, raw_document: RawDocument
    ) -> List[ResourceAccessPolicy]:
        return [AlwaysAllowAccessPolicy()]
